### ISSUE TITLE
RNN loss converging too early than expected.

### ISSUE THREAD

**Issue by Krishnkant-Swarnkar** [Krishnkant-Swarnkar/CS224d-Assignment-Solutions#1]
Recurrent Neural Network which I implemented for Assignment 2 Question 3 (Language Model) is expected to achieve a validation perplexity below 175. But My RNN-LM model converged to validation perplexity of 9035 even after about 4 hours of CPU training. I even used annealing learning rate (Exponential decay of Learning rate) to prevent overshooting.

**Comment by Krishnkant-Swarnkar** [comment]
The problem was because I applied the softmax function twice on the same hidden layer.
Now I've fixed it and it converges well.
